LINUX.md

```diff
@@ -0,0 +1,177 @@
+
+1.1. Download the VMime source used by the development team, from the link below:
+  - [vmime-0.9.2.tar.gz](http://www.dpi.inpe.br/terrama2-devel/vmime-0.9.2.tar.gz).
+  
+1.2. Open the shell command line and change the current directory to that folder:
+```
+$ cd /home/USER/mydevel/terrama2/codebase/packages/deb-package
+```
+
+1.3. Copy the downloaded source to current directory.
+
+1.4. In the shell command line, call the script *deb-vmime.sh* that will generate the debian package for VMime:
+```
+$ ./deb-vmime.sh
+```
+
+1.5. When the script finishes, just run the following command to install VMime package:
+```
+$ sudo dpkg -i terrama2-vmime_0.9.2-ubuntu16.04_amd64.deb
+```
+
+## Build Instructions
+
+After choosing the right branch or tag to work on, make sure you have all the third-party library dependencies needed before trying to build TerraMA².
+
+The `build/cmake` folder contains a CMake project for building TerraMA².
+
+You should use at least CMake version 2.8.12. Older versions than this may not work properly.
+
+### Development Environment
+
+1.1. Open a Command Prompt (Shell).
+
+1.2. Create a folder out of the TerraMA² source tree to generate the build system, for example:
+```
+$ cd /home/USER/mydevel/terrama2
+$ mkdir build-release
+$ cd build-release
+```
+**Note:** for the sake of simplicity create this directory in the same level as the source tree (as showned above).
+
+1.3. For Linux systems you must choose the build configuration:
+```
+$ cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE:STRING="Release" -DCMAKE_INSTALL_PREFIX:PATH="/home/USER/myinstall/terrama2" -DCMAKE_PREFIX_PATH:PATH="/home/USER/mylibs;/home/USER/mylibs/terralib5/lib/cmake" ../codebase/build/cmake
+```
+
+1.4. Building (with 4 process in parallel):
+```
+$ make -j 4
+```
+
+1.5. Installing:
+```
+$ make install
+```
+
+1.6. Uninstalling:
+```
+$ make uninstall
+```
+
+**Notes:** 
+
+- Some Linux flavours with different versions of GNU gcc and Boost will need more parameters such as:
+```
+  -DCMAKE_INCLUDE_PATH:PATH="/usr/local;/opt/include"
+  -DCMAKE_LIBRARY_PATH:PATH="/usr/local;/opt/lib"
+  -DCMAKE_PROGRAM_PATH:PATH="/usr/local/bin;/opt/bin"
+  -DBOOST_ROOT:PATH="/opt/boost"
+```
+
+- Boost can also be indicated by *BOOST_INCLUDEDIR*:
+```
+  -DBOOST_INCLUDEDIR:PATH="/usr/local/include"
+```
+
+- The parameter -lpthread must be informed only if your Boost was not built as a shared library:
+```
+  -DCMAKE_CXX_FLAGS:STRING="-lpthread"
+```
+
+- For building with Qt5 you can provide the *Qt5_DIR* variable as:
+```
+  -DQt5_DIR:PATH="/usr/local/lib/cmake/Qt5"
+```
+
+- For generating a debug version set *CMAKE_BUILD_TYPE* as:
+```
+  -DCMAKE_BUILD_TYPE:STRING="Debug"
+```
+
+### Package
+
+1.1 Change the current directory to that folder:
+```
+$ cd /home/USER/mydevel/terrama2/codebase/packages/deb-package
+```
+
+1.2. In the shell command line, call the script *deb-terrama2.sh* to build and generate automatically the TerraMA² package:
```

Comment:
note the script will assume installed terralib (and dependencies) at /opt/terralib/5.2.2